RocketChat/Rocket.Chat.ReactNative
# fix: disable backup of mmkv keys in icloud

## Proposed changes


Currently the mmkv library sets accessibility by [default](https://github.com/ammarahm-ed/react-native-mmkv-storage/blob/2aeabecd6a9c9092ea81e19a99cf853539dfe6f2/ios/SecureStorage.m#L253C31-L253C31) to when creating a value in the keychain, using the [`kSecAttrAccessibleAfterFirstUnlock`](https://developer.apple.com/documentation/security/ksecattraccessibleafterfirstunlock) property.
However, when restoring the backup, the library cannot update the backup data, since the keys are saved with an encryption that only stays on the device.
So when opening the app, she can't access the saved keys (thus having to login again) and also can't update the old keys (thus not saving the user session, having to login every time)

Updated the property so keys are only on the device and not backed up. [`kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly`](https://developer.apple.com/documentation/security/ksecattraccessibleafterfirstunlockthisdeviceonly)

And also in this PR we apply a patch so that the entire folder related to the mmkv backup is not sent to the backup as well, through the `addSkipBackupAttributeToURLAtPath` function that was injected via path package.

## Issue(s)	


## How to test or reproduce


- Restore an icloud backup on a new device and see if after login the session will remain active after closing the application.

## Screenshots

## Types of changes




## Checklist



## Further comments


Jira: [NATIVE-349]

[NATIVE-349]: https://rocketchat.atlassian.net/browse/NATIVE-349?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
